recipes/ninja/1.10.x/conanfile.py

```diff
@@ -0,0 +1,55 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class NinjaConan(ConanFile):
+    name = "ninja"
+    description = "Ninja is a small build system with a focus on speed"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ninja-build/ninja"
+    settings = "os", "arch", "compiler"
+    exports_sources = ["CMakeLists.txt", "*.patch"]
+    generators = "cmake"
+
+    _source_subfolder = "source_subfolder"
+    _cmake = None
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.compiler.runtime != "MT":
+                raise ConanInvalidConfiguration("Only MT MSVC runtime is supported")
```

Comment:
@robert-shade why is this ? it prevents a lot of consumers to use this package. Shouldn't this raise be in `def build(self)` so that all binaries always use MT, but consumer can use whatever runtime they need ?
(see also https://github.com/conan-io/conan-center-index/issues/1831)
CC @patricia-gallardo